integration/mediation-tests/tests-other/src/test/java/org/wso2/carbon/esb/handler/HandlerTest.java

```diff
@@ -0,0 +1,126 @@
+import java.io.File;
+import java.io.IOException;
+import java.net.URISyntaxException;
+
+/**
+ * Test class to test a synapse handler.
+ */
+public class HandlerTest extends ESBIntegrationTest {
+
+    private static final String JAR_NAME = "org.wso2.carbon.test.gateway-1.0.0.jar";
+    private static final String CONF_NAME = "synapse-handlers.xml";
+    private static final String LOCATION = "/artifacts/ESB/handler";
+
+    private ServerConfigurationManager serverConfigurationManager;
+    private LogViewerClient logViewerClient;
+
+    @BeforeClass(alwaysRun = true)
+    public void setEnvironment() throws Exception {
+        super.init();
+        serverConfigurationManager = new ServerConfigurationManager(context);
+        serverConfigurationManager
+                .copyToComponentDropins(new File(getClass().getResource(LOCATION + File.separator + JAR_NAME).toURI()));
+        copyToComponentConf(getClass().getResource(LOCATION + File.separator + CONF_NAME).getPath(), CONF_NAME);
+        serverConfigurationManager.restartForcefully();
+        super.init();
+        logViewerClient = new LogViewerClient(contextUrls.getBackEndUrl(), getSessionCookie());
+        logViewerClient.clearLogs();
+    }
+
+    @Test(groups = { "wso2.esb" },
+          description = "Sending a Message Via proxy to check synapse handler logs")
+    public void testSynapseHandlerExecution()
+            throws IOException, EndpointAdminEndpointAdminException, LoginAuthenticationExceptionException,
+            XMLStreamException, LogViewerLogViewerException {
+        boolean requestInStatus = false;
+        boolean requestOutStatus = false;
+        boolean responseInStatus = false;
+        boolean responseOutStatus = false;
+        boolean handlerStatus = false;
+
+        OMElement response = axis2Client
+                .sendSimpleStockQuoteRequest(getProxyServiceURLHttp("handlerTestProxy"), null, "WSO2");
+        Assert.assertNotNull(response);
+        LogEvent[] logs = logViewerClient.getAllRemoteSystemLogs();
+        for (LogEvent logEvent : logs) {
+            String message = logEvent.getMessage();
+            if (message.contains("handleRequestInFlow")) {
+                requestInStatus = true;
+            }
+
+            if (message.contains("handleRequestOutFlow")) {
+                requestOutStatus = true;
+            }
+
+            if (message.contains("handleResponseInFlow")) {
+                responseInStatus = true;
+            }
+
+            if (message.contains("handleResponseOutFlow")) {
+                responseOutStatus = true;
+            }
+        }
+
+        if (requestInStatus && requestOutStatus && responseInStatus && responseOutStatus) {
+            handlerStatus = true;
+        }
+
+        Assert.assertTrue(handlerStatus, "Synapse handler not working properly");
+
+    }
+
+    public void copyToComponentConf(String sourcePath, String fileName) throws IOException, URISyntaxException {
```

Comment:
Make` copyToComponentConf() `method private because it's only call within the class.

Reply:
Fixed with https://github.com/wso2/product-ei/pull/2148